lib/adapters/neo4j.js

```diff
@@ -0,0 +1,426 @@
+  }
+
+  schema = mongoose.Schema(schema, options);
+  schema.refkeys = refkeys;
+
+  _.each(refkeys, function(key){
+    var index = {};
+    index[key] = 1;
+
+    schema.index(index);
+  });
+
+  //Set index on deletedAt
+  schema.index({
+    deletedAt: 1
+  },{
+    sparse: true
+  });
+
+  if (schemaCallback)
+    schemaCallback(schema);
+
+  _schema[name] = schema;
+
+  return schema;
+
+  function typeCheck(fn) {
+    return Object.prototype.toString.call(new fn(''))
+      .slice(1, -1).split(' ')[1].toLowerCase();
+  }
+};
+
+adapter.model = function(name, schema, options) {
+
+  // console.log("neo4j model", name, schema, options);
+  // console.log(_.keys(_schema));
+
+  // if(schema) {
+  var model = _.clone(_schema[name]);
+    // this._models[name] = model;
+  return _.extend(model, options, { modelName : name});
+  // } else {
+  //   return this._models[name];
+  // }
+  // throw new Error("NotImplemented")
+};
+
+adapter.create = function (model, resource) {
+  var _this = this;
+
+  console.log("neo4j create", resource);
+
+  console.log("neo4j _this.db", _this.db);
+  
+  var node = _this.db.createNode(resource);
+
+  return new Promise(function (resolve, reject) {
+    
+    node.save(function (err, node) {    // ...this is what actually persists.
+        if (err) {
+          console.error('Error saving new node to database:', err);
+          reject(err);
+        } else {
+          console.log('Node saved to database with id:', node.id, node);
+          resolve([_.extend(node.data, { id : node.id })])
+        }
+    });
+
+    // that.db.getNodeById(query.id, function(err, node) {
+    //   // resolve([node]);
+    //   console.log("getNodeById callback", err.length, node);
+    //   resolve([{ id : query.id }])
+    // });
+    
+  });
+
+
+  return resource;
+  // throw new Error("NotImplemented")
+};
+
+adapter.update = function (model, id, update) {
+
+  console.log("neo4j update", model, id, update);
+  throw new Error("NotImplemented")
+};
+
+adapter.markDeleted = function(model, id){
+  console.log("neo4j markDeleted", model, id);
+  throw new Error("NotImplemented")
+};
+
+adapter.delete = function (model, id) {
+  console.log("neo4j create", model, id);
+  throw new Error("NotImplemented")
+};
+
+/**
+ *
+ * @param model {Model}
+ * @param query {Object}
+ * @param projection {Object}
+ * @returns {Promise}
+ */
+adapter.find = function(model, query, projection){
+  console.log("neo4j find", model, query, projection);
+  throw new Error("NotImplemented")
+};
+
+/**
+ * @param model {Model || String}
+ * @param query {Object}
+ * //@param limit {Number} - deprecated as unused
+ * @param projection {Object}
+ * @returns {Promise}
+ */
+
+adapter.findMany = function(model, query, projection) {
+  console.log("neo4j findMany", query, projection); //model, );
+  var that = this;
+
+  if (query.id) {
+    //Don't know why this is coming into findMany with id - would expect it to go into .find...
```

Comment:
There's complicated process of parsing a query and we shrunk `.find` to single line that adds additional `{limit: 1}` and passes query to `.findMany`. Also - there's no point feeding database with complex query if it has single id. This might happen in fortune hooks that modify clients query, e.g. security one.